#### Description
The Fitbit sleep tracking data consists of 67 entries, each containing sleep-related metrics. The dataset includes eight columns:

- 'Id' represents the unique identifier of each record.
- 'Date' denotes the date of the sleep data.
- 'WeightKg' indicates the weight of the user in kilograms.
- 'WeightPounds' indicates the weight of the user in pounds.
- 'Fat' represents the amount of body fat, although this column has only 2 non-null values.
- 'BMI' stands for Body Mass Index, which is a numeric value derived from weight and height.
- 'IsManualReport' is a boolean column indicating whether the report was entered manually.
- 'LogId' is the identifier associated with each record.

The data columns contain both numerical and categorical data types. The 'Date' column is of object type, which likely represents the date in a string format. The dataset seems to have some missing values, particularly in the 'Fat' column. The memory usage of the dataset is relatively low at 3.9+ KB.

This data can potentially offer insights into the relationship between sleep metrics and other health-related factors for users utilizing Fitbit's tracking features.

In [1]:
import pandas as pd
from ydata_profiling import ProfileReport

import warnings

%matplotlib inline

warnings.filterwarnings("ignore", category=FutureWarning)
warnings.filterwarnings("ignore", category=DeprecationWarning)

In [2]:
df_weight_log_info_merged = pd.read_csv('weightLogInfo_merged.csv')

In [3]:
df_weight_log_info_merged.head()

Id                   Date    WeightKg  WeightPounds   Fat  \
0  1503960366   5/2/2016 11:59:59 PM   52.599998    115.963147  22.0   
1  1503960366   5/3/2016 11:59:59 PM   52.599998    115.963147   NaN   
2  1927972279   4/13/2016 1:08:52 AM  133.500000    294.317120   NaN   
3  2873212765  4/21/2016 11:59:59 PM   56.700001    125.002104   NaN   
4  2873212765  5/12/2016 11:59:59 PM   57.299999    126.324875   NaN   

         BMI  IsManualReport          LogId  
0  22.650000            True  1462233599000  
1  22.650000            True  1462319999000  
2  47.540001           False  1460509732000  
3  21.450001            True  1461283199000  
4  21.690001            True  1463097599000

In [4]:
df_weight_log_info_merged.shape

(67, 8)

In [6]:
df_weight_log_info_merged.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 67 entries, 0 to 66
Data columns (total 8 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Id              67 non-null     int64  
 1   Date            67 non-null     object 
 2   WeightKg        67 non-null     float64
 3   WeightPounds    67 non-null     float64
 4   Fat             2 non-null      float64
 5   BMI             67 non-null     float64
 6   IsManualReport  67 non-null     bool   
 7   LogId           67 non-null     int64  
dtypes: bool(1), float64(4), int64(2), object(1)
memory usage: 3.9+ KB


In [7]:
df_weight_log_info_merged.describe()

Id    WeightKg  WeightPounds       Fat        BMI  \
count  6.700000e+01   67.000000     67.000000   2.00000  67.000000   
mean   7.009282e+09   72.035821    158.811801  23.50000  25.185224   
std    1.950322e+09   13.923206     30.695415   2.12132   3.066963   
min    1.503960e+09   52.599998    115.963147  22.00000  21.450001   
25%    6.962181e+09   61.400002    135.363832  22.75000  23.959999   
50%    6.962181e+09   62.500000    137.788914  23.50000  24.389999   
75%    8.877689e+09   85.049999    187.503152  24.25000  25.559999   
max    8.877689e+09  133.500000    294.317120  25.00000  47.540001   

              LogId  
count  6.700000e+01  
mean   1.461772e+12  
std    7.829948e+08  
min    1.460444e+12  
25%    1.461079e+12  
50%    1.461802e+12  
75%    1.462375e+12  
max    1.463098e+12

#### Summary Statistics Analysis

The summary statistics of the dataset reveal interesting insights about the various metrics:

- **WeightKg**: The weight of users ranges from a minimum of approximately 52.60 kg to a maximum of 133.50 kg. The average weight is about 72.04 kg, with a standard deviation of 13.92 kg. This indicates a relatively wide distribution of weights among the users.

- **WeightPounds**: The weight of users in pounds has a minimum of around 115.96 pounds and a maximum of 294.32 pounds. The mean weight in pounds, approximately 158.81, aligns with the mean weight in kilograms.

- **Fat**: The 'Fat' column has only 2 non-null values, making it difficult to draw meaningful conclusions from the summary statistics for this column.

- **BMI (Body Mass Index)**: The BMI values range from a minimum of 21.45 to a maximum of 47.54. The average BMI is about 25.19, with a standard deviation of 3.07. The variation in BMI suggests diversity in the users' body compositions.

These summary statistics provide a glimpse into the diversity and range of the dataset's metrics. It's worth noting that the range of BMI values indicates variability in users' body compositions, which reinforces the limitations of using BMI as a sole indicator of health due to its failure to consider factors like muscle mass and fat distribution.

Overall, these statistics highlight the heterogeneity of the user population and emphasize the importance of considering multiple factors when assessing health and fitness.

In [5]:
#profile = ProfileReport(df_weight_log_info_merged, title="Pandas Profiling Report")
#profile.to_widgets()